src/unix/netbsd.c

```diff
@@ -66,22 +66,34 @@ void uv_loadavg(double avg[3]) {
 
 
 int uv_exepath(char* buffer, size_t* size) {
+  /*
+   * Intermediate buffer, retrieving partial path name does not work
+   * As of NetBSD-8(beta), vnode->path translator does not handle files
+   * with longer names than 31 characters
+   */
+  char int_buf[PATH_MAX];
+  size_t int_size = __arraycount(int_buf);
+
   int mib[4];
-  size_t cb;
-  pid_t mypid;
 
   if (buffer == NULL || size == NULL || *size == 0)
     return -EINVAL;
 
-  mypid = getpid();
   mib[0] = CTL_KERN;
   mib[1] = KERN_PROC_ARGS;
-  mib[2] = mypid;
-  mib[3] = KERN_PROC_ARGV;
+  mib[2] = -1;
+  mib[3] = KERN_PROC_PATHNAME;
 
-  cb = *size;
-  if (sysctl(mib, 4, buffer, &cb, NULL, 0))
+  if (sysctl(mib, 4, int_buf, &int_size, NULL, 0))
     return -errno;
+
+  /*
+   * Copy string from the intermediate buffer to outer one with appropriate
+   * length
+   */
+  strlcpy(buffer, int_buf, *size);
```

Comment:
Is `int_buf` always nul-terminated?  What if the executable path is exactly PATH_MAX characters long?

Reply:
`int_buf` is always nul-terminated: https://nxr.netbsd.org/xref/src/sys/kern/vfs_getcwd.c#588

If I understand correctly, we actually populate the string up to `PATH_MAX/2` characters.

If something breaks in extreme conditions, we will fix it on the NetBSD side.

This interface is fragile right now and does not cache files with long file names and it will be at some point replaced with a better implementation.